File: app-java/src/main/java/tech/thdev/java_udemy_sample/view/image/ImageSampleFragment.java
package tech.thdev.java_udemy_sample.view.image;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;

import com.google.android.material.floatingactionbutton.FloatingActionButton;

import org.jetbrains.annotations.NotNull;

import tech.thdev.java_udemy_sample.R;
import tech.thdev.java_udemy_sample.adapter.ImageSampleAdapter;
import tech.thdev.java_udemy_sample.data.ImageItem;
import tech.thdev.java_udemy_sample.data.source.image.ImageSampleRepository;
import tech.thdev.java_udemy_sample.databinding.FragmentImageSampleBinding;
import tech.thdev.java_udemy_sample.view.image.presenter.ImageSamplePresenter;
import tech.thdev.java_udemy_sample.view.image.presenter.ImageSamplePresenterImpl;

/**
 * Created by tae-hwan on 10/29/16.
 */

public class ImageSampleFragment extends Fragment implements ImageSamplePresenter.View {

    private FragmentImageSampleBinding fragmentImageSampleBinding;

    // MVPPresenter 추가
    private ImageSamplePresenterImpl presenter;

    private ImageSampleAdapter adapter;

    // static instance 생성
    public static ImageSampleFragment getInstance() {
        return new ImageSampleFragment();
    }

    @Nullable
    @Override
    public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        fragmentImageSampleBinding = FragmentImageSampleBinding.inflate(inflater, container, false);
        return fragmentImageSampleBinding.getRoot();
    }

    @Override
    public void onViewCreated(@NotNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);

        presenter = new ImageSamplePresenterImpl(this, ImageSampleRepository.getInstance());

        adapter = new ImageSampleAdapter(getContext());
        fragmentImageSampleBinding.recyclerView.setAdapter(adapter);

        // Activity의 {@link FloatingActionButton}
        FloatingActionButton fab = (FloatingActionButton) getActivity().findViewById(R.id.fab);
        fab.setOnClickListener(new View.OnClickListener() {

            @Override
            public void onClick(View v) {
                presenter.updateImageData(getContext());
            }
        });

        // 새로운 데이터 불러오기
        presenter.updateImageData(getContext());
    }

    @Override
    public void notifyAdapter() {
        adapter.notifyDataSetChanged();
    }

    @Override
    public void addItem(ImageItem imageItem) {
        adapter.addItem(imageItem);
    }
}

File: app-kotlin/src/main/java/tech/thdev/kotlin_udemy_sample/view/image/ImageSampleFragment.kt
package tech.thdev.kotlin_udemy_sample.view.image

import android.os.Bundle
import android.util.Log
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Toast
import androidx.fragment.app.Fragment
import com.google.android.material.floatingactionbutton.FloatingActionButton
import tech.thdev.kotlin_udemy_sample.R
import tech.thdev.kotlin_udemy_sample.data.model.PhotoDataSource
import tech.thdev.kotlin_udemy_sample.databinding.FragmentImageSampleBinding
import tech.thdev.kotlin_udemy_sample.view.image.adapter.ImageAdapter
import tech.thdev.kotlin_udemy_sample.view.image.presenter.ImageContract
import tech.thdev.kotlin_udemy_sample.view.image.presenter.ImagePresenter

/**
 * Created by tae-hwan on 10/3/16.
 */
class ImageSampleFragment : Fragment(), ImageContract.View {

    private val fab by lazy {
        requireActivity().findViewById<FloatingActionButton>(R.id.fab)
    }

    // Java 식의 static instance
    companion object {
        fun getInstance() = ImageSampleFragment()
    }

    private var imageAdapter: ImageAdapter? = null

    private var presenter: ImageContract.Presenter? = null

    private lateinit var fragmentImageSampleBinding: FragmentImageSampleBinding

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View =
        FragmentImageSampleBinding.inflate(inflater, container, false).also {
            fragmentImageSampleBinding = it
        }.root

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)

        presenter = ImagePresenter()
        presenter?.view = this

        /**
         * Model을 생성하여 셋팅한다
         */
        presenter?.photoDataSample = PhotoDataSource

        imageAdapter = ImageAdapter(requireContext())
        fragmentImageSampleBinding.recyclerView.adapter = imageAdapter

        fab.setOnClickListener {
            presenter?.getRecentImageSample()
        }

        presenter?.getRecentImageSample()
    }

    override fun showLoadFailMessage(message: String) {
        Toast.makeText(context, message, Toast.LENGTH_LONG).show()
        Log.e("TAG", "Exception : " + message)
    }

    override fun showLoadSuccess() {
        Toast.makeText(context, "Load success", Toast.LENGTH_SHORT).show()
    }

    override fun showLoadFail() {
        Toast.makeText(context, "Load fail", Toast.LENGTH_SHORT).show()
    }
}